handler/src/main/java/io/netty/handler/ssl/SslHandler.java

```diff
@@ -1389,9 +1400,10 @@ public void operationComplete(Future<Channel> future) throws Exception {
         try {
             engine.beginHandshake();
             wrapNonAppData(ctx, false);
-            ctx.flush();
-        } catch (Exception e) {
-            notifyHandshakeFailure(e);
+        } catch (SSLException e) {
```

Comment:
either change this to `Throwable` or add another catch clause and re-throw to ensure we free our resources?